src/math_parser_diag.cpp

```diff
@@ -443,6 +523,190 @@ std::function<void( dialogue &, double )> spell_exp_ass( char scope,
     };
 }
 
+std::function<double( dialogue & )> spell_level_eval( char scope,
+        std::vector<diag_value> const &/* params */, diag_kwargs const &kwargs )
+{
+    diag_value school( std::string{} );
+    if( kwargs.count( "school" ) ) {
+        school = *kwargs.at( "school" );
+    }
+    diag_value spell( std::string{} );
+    if( kwargs.count( "spell" ) != 0 ) {
+        spell = *kwargs.at( "spell" );
+    }
```

Comment:
Consider splitting and flattening this to `u_spell_level()` and `u_school_level()` instead of using a mandatory kwarg.

Maybe @GuardianDll can chime in with their preference.

Reply:
I don't have strong preference to any of the side, but i lean to splitting because of it's shorter syntax

Reply:
I will split it into two parts: u_spell_level() and u_school_level().